doc/index.md

```diff
@@ -236,114 +292,131 @@ local myconnection = http_client.connect {
+		  - `1.1`
+		  - `2`
 
+	  - `h2_settings` (table, optional): HTTP 2 settings to use. See [*http.h2_connection*](#http.h2_connection) for details
 
-### `h1_connection:checktls()` <!-- --> {#http.h1_connection:checktls}
 
-See [`connection:checktls()`](#connection:checktls)
+## http.h1_connection
 
+The h1_connection module adheres to the [connection](#connection) and provides HTTP 1 and 1.1 specific operations.  
 
-### `h1_connection:localname()` <!-- --> {#http.h1_connection:localname}
+### `connection.new(socket, conn_type, version)` <!-- --> {#connection.new}
 
-See [`connection:localname()`](#connection:localname)
+Constructor for a new connection. Takes a socket instance, a connection type string and a numeric HTTP version number. Valid values for the connection type are "client" and "server". Valid values for the version number are 1 and 1.1. Success returns the newly initialized connection object in a non-connected state. Failure returns 
 
 
-### `h1_connection:peername()` <!-- --> {#http.h1_connection:peername}
+### `h1_connection.version` <!-- --> {#http.h1_connection.version}
 
-See [`connection:peername()`](#connection:peername)
+Used to specificy the http version used for the connection handshake. Valid values are:
+- `1.0` 
+- `1.1`
 
 
 ### `h1_connection:clearerr(...)` <!-- --> {#http.h1_connection:clearerr}
 
+ 
 
 ### `h1_connection:take_socket()` <!-- --> {#http.h1_connection:take_socket}
 
 
-### `h1_connection:shutdown(dir)` <!-- --> {#http.h1_connection:shutdown}
-
-See [`connection:shutdown()`](#connection:shutdown)
-
-
-### `h1_connection:close()` <!-- --> {#http.h1_connection:close}
-
-See [`connection:close()`](#connection:close)
-
-
-### `h1_connection:new_stream()` <!-- --> {#http.h1_connection:new_stream}
-
-See [`connection:new_stream()`](#connection:new_stream)
-
-
-### `h1_connection:get_next_incoming_stream(timeout)` <!-- --> {#http.h1_connection:get_next_incoming_stream}
-
-See [`connection:get_next_incoming_stream()`](#connection:get_next_incoming_stream)
-
-
 ### `h1_connection:flush(...)` <!-- --> {#http.h1_connection:flush}
 
+Flushes all buffered data on the socket. Returns true on success. Returns false and the error if the socket fails to flush.
+
 
 ### `h1_connection:read_request_line(timeout)` <!-- --> {#http.h1_connection:read_request_line}
 
+Gets the request method, requested path and HTTP version for an incoming request. On error, returns nil, an error message and an error number.
+
 
 ### `h1_connection:read_status_line(timeout)` <!-- --> {#http.h1_connection:read_status_line}
 
+Gets the HTTP version (1 or 1.1) , status code and a reason description if applicable. If the socket could not be read, returns nil, an error message and an error number.
+
 
 ### `h1_connection:read_header(timeout)` <!-- --> {#http.h1_connection:read_header}
 
+Returns the HTTP header (?key, val) from an incoming request. If the header could not be read, the function returns nil an error and an error message.
```

Comment:
Reads an http header from the underlying socket. On success, returns the key and the value as separate strings.